pkg/generators/openapi.go

```diff
@@ -648,13 +648,29 @@ func (g openAPITypeWriter) generateMapProperty(t *types.Type) error {
 		g.generateReferenceProperty(elemType)
 	case types.Slice, types.Array:
 		g.generateSliceProperty(elemType)
+	case types.Map:
+		g.generateSubMapLimitedSupport(t.Name, elemType)
 	default:
 		return fmt.Errorf("map Element kind %v is not supported in %v", elemType.Kind, t.Name)
 	}
 	g.Do("},\n},\n},\n", nil)
 	return nil
 }
 
+func (g openAPITypeWriter) generateSubMapLimitedSupport(parentName types.Name, t *types.Type) error {
```

Comment:
what would the generic solution look like?

Reply:
@stts You are right. Solution looks more generic now.